netstandard/tools/HandlePackageFileConflicts.netstandard.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Utilities;
+using System;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public partial class HandlePackageFileConflicts : Task
+    {
+        private Version GetAssemblyVersion(string sourcePath)
+        {
+            using (var assemblyStream = new FileStream(sourcePath, FileMode.Open, FileAccess.Read, FileShare.Delete | FileShare.Read))
+            {
+                Version result = null;
+                try
+                {
+                    using (PEReader peReader = new PEReader(assemblyStream, PEStreamOptions.LeaveOpen))
+                    {
+                        if (peReader.HasMetadata)
+                        {
+                            MetadataReader reader = peReader.GetMetadataReader();
+                            if (reader.IsAssembly)
+                            {
+                                result = reader.GetAssemblyDefinition().Version;
+                            }
+                        }
+                    }
+                }
+                catch (BadImageFormatException)
```

Comment:
HasMetatadata does.  Suppose its a corrupt dll, or any random file.  You'll hit this.  I just gave a try with a random text file named .dll and I hit 
```
   at System.Reflection.Throw.ImageTooSmall()
   at System.Reflection.PortableExecutable.PEBinaryReader.CheckBounds(UInt32 count)
   at System.Reflection.PortableExecutable.PEBinaryReader.ReadUInt16()
   at System.Reflection.PortableExecutable.PEHeaders.SkipDosHeader(PEBinaryReader& reader, Boolean& isCOFFOnly)
   at System.Reflection.PortableExecutable.PEHeaders..ctor(Stream peStream, Int32 size, Boolean isLoadedImage)
   at System.Reflection.PortableExecutable.PEReader.ReadPEHeadersNoLock(Stream stream, Int64 imageStartPosition, Int32 imageSize, Boolean isLoadedImage)
   at System.Reflection.PortableExecutable.PEReader.InitializePEHeaders()
   at System.Reflection.PortableExecutable.PEReader.get_PEHeaders()
   at System.Reflection.PortableExecutable.PEReader.get_HasMetadata()
```